functions/pom.xml

```diff
@@ -0,0 +1,47 @@
+<!--
+  Copyright 2020 Google LLC
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>com.example.cloud.functions</groupId>
+  <artifactId>functions-snippets</artifactId>
+  <version>1.0.0-SNAPSHOT</version>
+
+  <properties>
+    <maven.compiler.target>11</maven.compiler.target>
+    <maven.compiler.source>11</maven.compiler.source>
+    <kokoro.skipTests>true</kokoro.skipTests> <!-- test children POMs instead -->
+  </properties>
+
+  <!--
+    List subfolders as separate modules, to enable top-level test runs
+    This POM should *not* be inherited by the per-function POMs
+  -->
+  <packaging>pom</packaging>
+  <modules>
+    <module>helloworld</module>
+    <module>http</module>
+    <module>imagemagick</module>
+    <module>slack</module>
+    <module>concepts</module>
+    <module>logging</module>
+    <module>firebase</module>
+    <module>ocr</module>
+  </modules>
+</project>
```

Comment:
Why do we have this?  How does it help users?

Reply:
this helps me to build and test all functions in one shot. it also allows me to open up functions in the IDE in one shot if i need to.  like i mentioned separately, each sub directory/module is still completely independent.

the parent in the children pom references a published pom, i feel it's good as long as a comment on it's not needed for user.

Reply:
@saturnism and I discussed this on Friday - this parent POM lets us run `mvn` builds across _all_ the samples to verify that _all of them_ work as intended.

(Otherwise, we'd have to run each one individually.)

Reply:
also discussed in the doc and the call, this causes no harm for the samples.

Reply:
I'll defer to @lesv for the final decision, but I'm still -1 on this (mostly because I don't like snowflakes). 

My concerns are largely mitigated since Ace has already adapted the script + left helpful comments in it and Ray made a good point about it being a one way configuration. However, I don't think we really expect users to run _all_ the tests in a single shot, which means this is only really for us. Our CI doesn't use this, which means there is nothing really stopping someone new adding a sample and not putting it here. So I guess I'd still prefer not to have it, but I'll leave it up to lesv@.

Reply:
Why not do what we do and just put a script that looks for `pom.xml` files and builds them.  For example, see [this one](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/.kokoro/tests/run_tests.sh).

Reply:
I don't really prefer we commit a new script either - parent pom for _personal_ convince makes sense, I just would prefer we didn't commit it in the repo.

Reply:
pls send data on the csat comment. the top level pom.xml doesn't need to be included in the doc.  i'm curious to see where the confusions were and whether it was because of other issues such as children pom inheriting this, which causes build failures. that in itself will be very frustrating.

Reply:
It was over 18m ago.

Reply:
BTW - it's possible to do this in `bash`, but it's not exactly the most straightforward thing (mainly because of array-vs-string gotchas):

```
find */pom.xml | xargs -I {} echo $(pwd)/{} | xargs -I {} dirname {} | xargs -I {} sh -c "cd {} && mvn clean verify"
```

Kurtis / Les - would you be open to keeping this parent POM if we added a comment to it like this?

```
This pom.xml file is kept for testing/Maven compatibility purposes only.

Please DO NOT add dependencies and/or common settings here. The child folders inherit from our `shared-configuration` stored on GitHub instead of this file; cross-POM changes should be added there.
```

Reply:
FWIW I think this is simpler:

```
for pom in $(find * -name pom.xml); do (cd $(dirname $pom) && mvn clean verify); done
```

Reply:
@ace-n The problem is we don't expect the engineers who follow / maintain to continue to maintain this.  We used to do it.  We stopped because it got overwhelming as the # of samples increased.

Reply:
Les, thanks for clarifying.

Since the modules are explicitly specified, I'm willing to drop this in favor of a programmatic (e.g. `bash`) solution.